python/cuml/experimental/explainer/base.pyx

```diff
@@ -0,0 +1,406 @@
+        model. If not specified, the explainer will try to get the dtype
+        of the model, if it cannot be queried, then it will default to
+        np.float32.
+    output_type : 'cupy' or 'numpy' (default = None)
+        Parameter to specify the type of data to output.
+        If not specified, the explainer will try to see if model is gpu based,
+        if so it will be set to `cupy`, otherwise it will be set to `numpy`.
+        For compatibility with SHAP's graphing libraries, specify `numpy`.
+
+    """
+
+    def __init__(self,
+                 *,
+                 model,
+                 background,
+                 order=None,
+                 order_default='C',
+                 link='identity',
+                 verbose=False,
+                 random_state=None,
+                 is_gpu_model=None,
+                 handle=None,
+                 dtype=None,
+                 output_type=None):
+
+        if verbose is True:
+            self.verbose = logger.level_debug
+        elif verbose is False:
+            self.verbose = logger.level_error
+        else:
+            self.verbose = verbose
+
+        if self.verbose >= logger.level_debug:
+            self.time_performance = True
+        else:
+            self.time_performance = False
+
+        if handle is None:
+            self.handle = get_handle_from_cuml_model_func(model,
+                                                          create_new=True)
+        else:
+            self.handle = handle
+
+        if order is None:
+            self.order = get_tag_from_model_func(func=model,
+                                                 tag='preferred_input_order',
+                                                 default=order_default)
+        else:
+            self.order = order
+
+        self.link = link
+        self.link_fn = get_link_fn_from_str_or_fn(link)
+        self.model = model
+        if is_gpu_model is None:
+            # todo (dgd): when sparse support is added, use this tag to see if
+            # model can accept sparse data
+            self.is_gpu_model = \
+                get_tag_from_model_func(func=model,
+                                        tag='X_types_gpu',
+                                        default=None) is not None
+        else:
+            self.is_gpu_model = is_gpu_model
+
+        # we are defaulting to numpy for now for compatibility
+        if output_type is None:
+            # self.output_type = 'cupy' if self.is_gpu_model else 'numpy'
+            self.output_type = 'numpy'
+        else:
+            self.output_type = output_type
+
+        # if not dtype is specified, we try to get it from the model
+        if dtype is None:
+            self.dtype = get_dtype_from_model_func(func=model,
+                                                   default=np.float32)
+        else:
+            if dtype in [np.float32, np.float64]:
+                self.dtype = np.dtype(dtype)
```

Comment:
out of curiosity why do you have to convert to np.dtype?

Reply:
I was doing the wrong order of things, I use the `dtype` function of numpy so that we accept string description of the dtypes without additional work